pysteps/postprocessing/probmatching.py

```diff
@@ -259,6 +260,57 @@ def _get_error(scale):
     return shift, scale, R.reshape(shape)
 
 
+def resample_distributions(first_array, second_array, probability_first_array):
+    """
+    Merges two distributions (e.g. from the extrapolation nowcast and NWP in the blending module)
+    to effectively combine two distributions for the probability matching without losing extremes.
+    Parameters
+    ----------
+    first_array: array_like
+        One of the two arrays from which the distribution should be samples (e.g. the extrapolation
+        cascade).
+    second_array: array_like
+        One of the two arrays from which the distribution should be samples (e.g. the NWP (model)
+        cascade).
+    probability_first_array: float
+        The weight that first_array should get (as a value between 0 and 1). This is
+        typically based on cascade level 2 (as is done for all the probability matching steps in `blending/steps.py`)
+
+    Returns
+    ----------
+    csort: array_like
+        The output distribution as first_array drawn binomial distribution from the input arrays first_array and second_array.
+    """
+    # Make sure that the probability is never: <0, >1 or nan
+    if probability_first_array < 0.0:
+        probability_first_array = 0.0
+    elif probability_first_array > 1.0:
+        probability_first_array = 1.0
+    elif np.isnan(probability_first_array):
+        probability_first_array = 0.0
```

Comment:
why this? if it can happen that the level 2 cascade weight is nan, this special case should be handled in `blending/steps.py` imho

Reply:
Your change to `probability_first_array = np.clip(probability_first_array, 0.0, 1.0)` is neat, better to do it that way. The nan was an extra check, but is (I think) not necessary anymore, as nans should not take place in the blending code (but weights above or below 1 and 0 are in theory possible).